--- a/neo/idlib/math/Simd_AltiVec.h
+++ b/neo/idlib/math/Simd_AltiVec.h
@@ -95,9 +95,9 @@
 
 // This turns on support for PPC intrinsics in the SIMD_AltiVec.cpp file. Right now it's only used for frsqrte. GCC
 // supports these intrinsics but XLC does not.
-#if defined(__GNUC__) && defined(__ALTIVEC__)
-    #define PPC_INTRINSICS
-#endif
+//#if defined(__GNUC__) && defined(__ALTIVEC__)
+//    #define PPC_INTRINSICS
+//#endif
 
 // This assumes that the idDrawVert array that is used in DeriveUnsmoothedTangents is aligned. If its not aligned,
 // then we don't get any speedup

